src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -312,47 +377,58 @@ export class MeetingManager implements AudioVideoObserver {
-      this.audioInputDevices.length
-    ) {
-      this.selectedAudioInputDevice = this.audioInputDevices[0].deviceId;
-      try {
-        await this.audioVideo?.chooseAudioInputDevice(
-          this.audioInputDevices[0].deviceId
-        );
-      } catch (error) {
-        console.error(`Error in selecting audio input device - ${error}`);
-      }
-      this.publishSelectedAudioInputDevice();
-    }
-    if (
-      !this.selectedAudioOutputDevice &&
-      this.audioOutputDevices &&
-      this.audioOutputDevices.length
-    ) {
-      this.selectedAudioOutputDevice = this.audioOutputDevices[0].deviceId;
-      if (supportsSetSinkId()) {
-        try {
-          await this.audioVideo?.chooseAudioOutputDevice(
-            this.audioOutputDevices[0].deviceId
-          );
-        } catch (error) {
-          console.error('Failed to choose audio output device.', error);
+  async listAndSelectDevice(device: Device): Promise<void> {
+    await this.updateDeviceList(device)
+    
+    switch (device) {
+      case Device.AudioInput:
+        if (
+          !this.selectedAudioInputDevice &&
+          this.audioInputDevices &&
+          this.audioInputDevices.length
+        ) {
+          this.selectedAudioInputDevice = this.audioInputDevices[0].deviceId;
+          try {
+            await this.audioVideo?.chooseAudioInputDevice(
+              this.audioInputDevices[0].deviceId
+            );
+          } catch (error) {
+            console.error(`Error in selecting audio input device - ${error}`);
+          }
+          this.publishSelectedAudioInputDevice();
         }
-      }
-      this.publishSelectedAudioOutputDevice();
-    }
-    if (
-      !this.selectedVideoInputDevice &&
-      this.videoInputDevices &&
-      this.videoInputDevices.length
-    ) {
-      this.selectedVideoInputDevice = this.videoInputDevices[0].deviceId;
-      this.publishSelectedVideoInputDevice();
+        break;
+      case Device.AudioOutput:
+        if (
+          !this.selectedAudioOutputDevice &&
+          this.audioOutputDevices &&
+          this.audioOutputDevices.length
+        ) {
+          this.selectedAudioOutputDevice = this.audioOutputDevices[0].deviceId;
+          if (supportsSetSinkId()) {
+            try {
+              await this.audioVideo?.chooseAudioOutputDevice(
+                this.audioOutputDevices[0].deviceId
+              );
+            } catch (error) {
+              console.error('Failed to choose audio output device.', error);
+            }
+          }
+          this.publishSelectedAudioOutputDevice();
+        }
+        break;
+      case Device.VideoInput:
+        if (
+          !this.selectedVideoInputDevice &&
+          this.videoInputDevices &&
+          this.videoInputDevices.length
+        ) {
+          this.selectedVideoInputDevice = this.videoInputDevices[0].deviceId;
+          this.publishSelectedVideoInputDevice();
+        }
+        break;
+      default:
+        break;
```

Comment:
nit: You do not need break for default. Also default should never happen right. If so, you should log or throw an error here.

Reply:
Throw error could be a breaking change here. I would prefer log something here. Could I just use `console.log`?

Reply:
Sure you can use console.error